packages/paste-website/src/components/site-wrapper/site-header/index.tsx

```diff
@@ -0,0 +1,19 @@
+import * as React from 'react';
+import {SiteHeaderDesktop} from './SiteHeaderDesktop';
+import {SiteHeaderMobile} from './SiteHeaderMobile';
+import {useWindowSize} from '../../../hooks/useWindowSize';
+
+export const SiteHeader: React.FC = () => {
+  const [value, setValue] = React.useState('');
+  const {breakpointIndex} = useWindowSize();
+
+  return (
+    <>
+      {breakpointIndex !== undefined && breakpointIndex <= 1 ? (
```

Comment:
Overall I think this approach is interesting but the flash of content is pretty bad experience wise.

I think wrapping these in Boxes and just using a responsive display prop would result is a better page experience

Reply:
100% agree. I've resolved the FoUC by rendering both the mobile and desktop components while the breakpointIndex is still undefined (i.e.: SSR and while rehydrating), and relying on CSS to show the right thing.  Once the page is rehydrated and all the JS starts to kick in, I go back to only rendering the elements that should be displayed. Best of both worlds!